keep/api/routes/settings.py

```diff
@@ -351,23 +351,115 @@ def _print_debug(self, *args):
             super()._print_debug(*args)
 
 
-@router.get("/apikey")
-def get_api_key(
+@router.post("/apikey", description="Create API key")
+async def create_key(
+    request: Request,
+    authenticated_entity: AuthenticatedEntity = Depends(
+        AuthVerifier(["write:settings"])
+    ),
+    session: Session = Depends(get_session),
+):
+    try:
+        body = await request.json()
+        unique_api_key_id = body['name']
+    except Exception:
+        raise HTTPException(status_code=400, detail="Invalid request body")
+
+    api_key = create_api_key(
+        session=session,
+        tenant_id=authenticated_entity.tenant_id,
+        created_by=authenticated_entity.email,
+        unique_api_key_id=unique_api_key_id,
+        role=AdminRole,
```

Comment:
the role should be taken from the request and not be static

Reply:
Fix: f72cdc80c1540a13e0d365ca0c5f59b9e59dabbe